src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1187,8 +1187,11 @@ class Global(var currentSettings: Settings, reporter0: LegacyReporter)
       }
       // Create phases and link them together. We supply the previous, and the ctor sets prev.next.
       val last  = components.foldLeft(NoPhase: Phase)((prev, c) => c newPhase prev)
-      // rewind (Iterator.iterate(last)(_.prev) dropWhile (_.prev ne NoPhase)).next
-      val first = { var p = last ; while (p.prev ne NoPhase) p = p.prev ; p }
+      val phaseList = Iterator.iterate(last)(_.prev).takeWhile(_ != NoPhase).toList.reverse
+      val minId = phaseList.map(_.id).min
+      val maxId = phaseList.map(_.id).max
+      nextFrom = Array.tabulate(maxId)(i => infoTransformers.nextFrom(i))
```

Comment:
I haven't looked how ids are allocated, but maybe we should make sure they are not (in some future) assigned randomly :-)

Reply:
It's just `val id: Id = if (prev eq null) 0 else prev.id + 1`, which seems unlikely to drift.